src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -824,19 +829,24 @@ public ThreadPoolWorkQueueThreadLocals(ThreadPoolWorkQueue tpq)
             threadLocalCompletionCountObject = ThreadPool.GetOrCreateThreadLocalCompletionCountObject();
         }
 
+        public void TransferLocalWork()
+        {
+            object? cb;
+            while ((cb = workStealingQueue.LocalPop()) != null)
+            {
+                Debug.Assert(null != cb);
+                workQueue.Enqueue(cb, forceGlobal: true);
+            }
+        }
+
         ~ThreadPoolWorkQueueThreadLocals()
         {
             // Transfer any pending workitems into the global queue so that they will be executed by another thread
             if (null != workStealingQueue)
             {
                 if (null != workQueue)
                 {
-                    object? cb;
-                    while ((cb = workStealingQueue.LocalPop()) != null)
-                    {
-                        Debug.Assert(null != cb);
-                        workQueue.Enqueue(cb, forceGlobal: true);
-                    }
+                    TransferLocalWork();
```

Comment:
In what situation will workQueue be null?

Reply:
Doesn't look like it can be, will fix